scripts/analysis/compare_grid_detection.py

```
@@ -0,0 +1,436 @@
+        success = grid_size is not None and grid_size > 0
+
+        return {
+            "method": "Brightness Analysis",
+            "grid_size": grid_size,
+            "confidence": confidence,
+            "cols": cols,
+            "rows": rows,
+            "vertical_detections": vertical_detections,
+            "horizontal_detections": horizontal_detections,
+            "success": success,
+        }
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)